docs/integration/KafkaIntegration.md

```diff
@@ -8,104 +8,164 @@ sidebar_position: 1
-  own convention for association between schema and topic: schema name should be in format: CamelCasedTopicNameKey for keys and CamelCasedTopicNameValue for values.
-  For example for `input-events` topic, schema name should be named `InputEventsKey` for key or `InputEventsValue` for value. Be aware that it may require change of schema name
-  not only in Azure portal but also inside schema content - those names should be the same to make serialization works correctly
-  
-### Schema and payload types
+To communicate with running Nussknacker's scenarios via Kafka sources and sinks you can use for example standard kafka-cli commands
+like `kafka-console-consumer`, `kafka-console-producer`, [kcat](https://github.com/edenhill/kcat), Confluent's
+`kafka-avro-console-consumer`, `kafka-avro-console-producer` commands for Confluent-based Avro encoded messages or graphical tools like [AKHQ](https://akhq.io).
+
+Be aware that Azure-based Avro encoded messages have a little different format than Confluent once - Schema ID is passed in headers instead of payload.
+It can be less supported by some available tools. See [Schema Registry comparison section](#schema-registry-comparison) for details.
+
+### Message Payload
 
 By default, Nussknacker supports two combinations of schema type and payload type:
 
-* Avro schema + Avro payload
-* JSON schema + JSON payload
+* Avro schema + Avro payload (binary format)
+* JSON schema + JSON payload (human readable, text format)
+
+Avro payloads are more concise, because messages contain only values and schema id - without information about message structure like field names.
 
-If you prefer using JSON payload with Avro schema, you can use `avroAsJsonSerialization` configuration setting to change that behaviour ([see Configuration for details](/docs/installation_configuration_guide/ModelConfiguration#common-kafka-configuration)).
+Avro payload is compatible with standard Kafka serializers and deserializers delivered by Schema Registry providers. Thanks to that you should be able to send messages to Nussknacker and read messages
+produced by Nussknacker using standard tooling available around those Schema Registries. To see how those formats are different, take a look at [Schema Registry comparison section](#schema-registry-comparison)
+
+For some situations it might be helpful to use JSON payload with Avro schema. Especially when your Schema Registry doesn't support JSON schemas.
+You can do that by enabling `avroAsJsonSerialization` configuration setting.
 
 ### Schema ID
 
-Nussknacker supports schema evolution.
+Each topic can contain messages written using different schema versions. Schema versions are identified by *Schema ID*. 
+Nussknacker need to know what was the schema used during writing to make message validation and schema evolution possible.
+Because of that it need to extract *Schema ID* from the message.
```

Comment:
it need -> Nussknacker needs

Reply:
Nussknacker will be used twice but maybe it is not the problem, I'll change it.